networkx/algorithms/approximation/contiguous_oriented_labeling.py

```diff
@@ -0,0 +1,279 @@
+                        if check_path(nodes_in_ears, path, x, (x, y)) and path != [
+                            x,
+                            y,
+                            x,
+                        ]:
+                            for i in range(len(path) - 1):
+                                if (path[i], path[i + 1]) not in ordered_edges:
+                                    if x == u:
+                                        ordered_edges.insert(
+                                            idx, (path[i], path[i + 1])
+                                        )
+                                        idx += 1
+                                        oriented_graph.add_edge(path[i], path[i + 1])
+                                    else:
+                                        if t_update:
+                                            x_directed = oriented_graph.in_edges(x)
+                                            for e1 in x_directed:
+                                                if (
+                                                    e1 in ordered_edges
+                                                    or tuple(reversed(e1))
+                                                    in ordered_edges
+                                                ):
+                                                    x_idx = min(
+                                                        ordered_edges.index(e1), x_idx
+                                                    )
+                                                    t = x_idx + 1
+                                        ordered_edges.insert(t, (path[i], path[i + 1]))
+                                        oriented_graph.add_edge(path[i], path[i + 1])
+                                        t += 1
+                                        t_update = False
+
+                cycle = False
+            elif (
+                edge[0] in nodes_in_ears and edge[1] in nodes_in_ears
+            ):  # if both of the vertices of the edge already in an ear we add it after the first edge directed into x
+                x = edge[0]
+                y = edge[1]
+                if t_update:
+                    x_directed = oriented_graph.in_edges(x)
+                    for e1 in x_directed:
+                        if e1 in ordered_edges or tuple(reversed(e1)) in ordered_edges:
+                            x_idx = min(ordered_edges.index(e1), x_idx)
+                            t = x_idx + 1
+                ordered_edges.insert(
+                    t, (x, y)
+                )  # append after the first edge directed into x
+                oriented_graph.add_edge(x, y)
+                t += 1
+                t_update = False
+
+    return ordered_edges, oriented_graph.in_edges
+
+
+def almost_bridgeless_to_bridgless(g):
+    """
+    This function takes an almost bridgeless graph and adds an edge so it makes it bridgeless
```

Comment:
Add explanations and examples. See here: https://github.com/networkx/networkx/pull/5585/files  how good examples look like.